File: Easy/Python/496. Next Greater Element I.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        ans = [0] * len(nums1)

        for i in range(len(nums1)):
            found = False
            for j in range(len(nums2)):
                if nums1[i] == nums2[j]:
                    found = True
                if found and nums1[i] < nums2[j]:
                    ans[i] = nums2[j]
                    break
            if ans[i] == 0:
                ans[i] -= 1
        
        return ans

File: Easy/Java/496. Next Greater Element I.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int[] ans = new int[nums1.length];

        for(int i = 0; i < nums1.length; i++){
            boolean found = false;
            for(int j = 0; j < nums2.length; j++){
                if(nums1[i] == nums2[j])
                    found = true;
                if(found && nums1[i] < nums2[j]){
                    ans[i] = nums2[j];
                    break;
                }
            }
            if(ans[i] == 0)
                ans[i]--;
        }

        return ans;
    }
}
